ingestion/src/metadata/ingestion/lineage/sql_lineage.py

```diff
@@ -163,6 +171,116 @@ def get_table_fqn_from_query_name(
     return database_query, schema_query, table
 
 
+def handle_udf_column_lineage(
+    column_lineage_original: dict,
+    column_lineage_generated: List[Tuple[Column, Column]],
+):
+    """
+    Handle UDF column lineage
+    """
+    try:
+        result = {}
+        intermediate_column_lineage = populate_column_lineage_map(
+            column_lineage_generated
+        )
+        # Iterate through the original dictionary
+        for source_table, mappings in column_lineage_original.items():
+            result[source_table] = {}
+            for intermediate_table, column_pairs in mappings.items():
+                # Iterate through each column mapping in the original dictionary
+                for source_column, intermediate_column in column_pairs:
+                    if intermediate_table in intermediate_column_lineage:
+                        # Check intermediate dictionary for mappings
+                        for (
+                            target_table,
+                            target_mappings,
+                        ) in intermediate_column_lineage[intermediate_table].items():
+                            for inter_col, target_col in target_mappings:
+                                if intermediate_column == inter_col:
+                                    # Append to the result dictionary
+                                    if target_table not in result[source_table]:
+                                        result[source_table][target_table] = []
+                                    result[source_table][target_table].append(
+                                        (source_column, target_col)
+                                    )
```

Comment:
How do you think about extracting the last for loop into its own methods to reduce the nesting

```
                            for inter_col, target_col in target_mappings:
                                if intermediate_column == inter_col:
                                    # Append to the result dictionary
                                    if target_table not in result[source_table]:
                                        result[source_table][target_table] = []
                                    result[source_table][target_table].append(
                                        (source_column, target_col)
                                    )
```